packages/esm-patient-clinical-view-app/src/clinical-view-form/clinical-view-form.component.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useEffect } from "react";
+import Search from "carbon-components-react/es/components/Search";
+import Checkbox from "carbon-components-react/es/components/Checkbox";
+import Button from "carbon-components-react/es/components/Button";
+import styles from "./clinical-view-form.component.scss";
+import debounce from "lodash-es/debounce";
+import { useTranslation } from "react-i18next";
+import {
+  detach,
+  temporaryConfigStore,
+  TemporaryConfigStore,
+  useConfig,
+} from "@openmrs/esm-framework";
+import { useClinicalView } from "../store";
+import isEmpty from "lodash-es/isEmpty";
+import cloneDeep from "lodash-es/cloneDeep";
+import { set } from "lodash-es";
+
+interface ClinicalViewFormProps {}
+interface View {
+  slotName: string;
+  slot: string;
+}
+
+const ClinicalViewForm: React.FC<ClinicalViewFormProps> = () => {
+  const { t } = useTranslation();
+  const { clinicalViews } = useConfig();
+  const moduleName = "@openmrs/esm-patient-clinical-view-app";
+  const path = [moduleName, "clinicalViews"];
+  const [searchTerm, setSearchTerm] = React.useState<string>("");
+  const [searchResults, setSearchResult] = React.useState<Array<View>>([]);
+  const [views, setViews] = React.useState<Array<View>>(useClinicalView());
+  const [tempConfig, setTempConfig] = useState<TemporaryConfigStore>(null);
+
+  React.useEffect(() => {
+    !isEmpty(views) && setSearchResult(views);
+  }, [views]);
+
+  React.useEffect(() => {
+    const results = views.filter(
+      (view) =>
+        view.slotName
+          .toLocaleLowerCase()
+          .search(searchTerm.toLocaleLowerCase()) !== -1
+    );
+    setSearchResult(results);
+  }, [searchTerm, views]);
+
+  const handleSearch = debounce((searchTerm) => {
+    setSearchTerm(searchTerm);
+  }, 300);
+
+  const handleChange = (slot: string, slotName: string, checked: boolean) => {
```

Comment:
Almost all of these functions (handleChange, closeClinicalViewForm, addClinicalView, ...) should be cached via `React.useCallback`.